### ISSUE TITLE
"qpdf --check" fails to detect multiply defined metadata

### ISSUE THREAD

**Issue by vinc17fr** [qpdf/qpdf#345]
PDF files must not have multiply defined metadata (as they are handle differently by different tools), and qpdf does not detect such errors.

```
$ qpdf --check metadata.pdf
checking metadata.pdf
PDF Version: 1.5
File is not encrypted
File is not linearized
No syntax or stream encoding errors found; the file may still contain
errors that qpdf cannot detect
```
on the file [metadata.pdf](https://github.com/qpdf/qpdf/files/3446653/metadata.pdf), which contains: `/Title (Metadata Test) /Author()/Title()/Subject()/Creator(LaTeX with hyperref)/Producer(pdfTeX-1.40.20)/Keywords()` i.e. with 2 Title fields.

**Comment by jberkenbilt** [comment]
Reminder to self: this is also https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=933397; be sure to close the debian bug when the release with this fix is uploaded.

**Comment by jberkenbilt** [comment]
At present, qpdf does not detect or even have a way to represent dictionaries with duplicated keys. I should be able to modify the parser to detect and warn for duplicated keys. Right now, qpdf would silently overwrite a dictionary entry if the key was duplicated. Unfortunately a fix for this will not make 9.0.0, but it can go in soon.

**Comment by jberkenbilt** [comment]
This will be fixed in 9.0.1, which I will release within the next few days.
